modules/packerina/src/main/java/org/ballerinalang/packerina/UserDirExtensionPackageRepositoryProvider.java

```diff
@@ -0,0 +1,27 @@
+package org.ballerinalang.packerina;
+
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.repository.PackageRepository;
+import org.ballerinalang.repository.fs.LocalFSPackageRepository;
+import org.ballerinalang.spi.ExtensionPackageRepositoryProvider;
+
+import java.nio.file.Path;
+
+import static org.ballerinalang.util.BLangConstants.USER_REPO_ARTIFACTS_DIRNAME;
+import static org.ballerinalang.util.BLangConstants.USER_REPO_SRC_DIRNAME;
+
+/**
+ *
+ * Load bal files form $HOME/.ballerina.
+ *
+ * ie: import foo.bar; will load from ~/.ballerina/artifacts/src/foo/bar/*.bal
+ */
+@JavaSPIService("org.ballerinalang.spi.ExtensionPackageRepositoryProvider")
+public class UserDirExtensionPackageRepositoryProvider implements ExtensionPackageRepositoryProvider {
```

Comment:
@sameera-jayasoma The way implemented here is as an "Extension" package repository provider, which is in the same level as other extensions we copy into the system. Our initial design had an "User" package repository concept, where the extension repository is the parent repo of the user repository; ref: https://github.com/manuranga/ballerina/blob/0357faf7b1da8c20fad016bdd328095d9110c1c0/modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/PackageLoader.java#L204. Which approach should we follow? ..

Reply:
Added a UserRepositoryProvider that has higher presidency than extension repo.